### PR TITLE
Fix incompatibility with z-moves, add migration command from aaron'x mod

### PR DESCRIPTION
**Pull request by landon-wills**
- Removes mixin used to add mega evolution, uses Cobblemon provided API instead
- Updates build process to use cobblemon 1.5.2
- Adds command to migrate held megastones from AaronMusks' mod to this one

### CODE REVIEW

File: common/src/main/java/com/selfdot/cobblemonmegas/common/CobblemonMegas.java [hunk 1]
```diff
@@ -105,6 +107,8 @@ private void onServerStarting(MinecraftServer server) {
 
     private Unit onBattleStartedPre(BattleStartedPreEvent event) {
         MegaUtils.deMegaEvolveAllPlayers(event.getBattle());
+        var players = event.getBattle().getPlayers();
+        players.forEach(player -> Cobblemon.playerData.get(player).getKeyItems().add(new Identifier("cobblemon", "key_stone")));
```
Comment (landon-wills) [inline common/src/main/java/com/selfdot/cobblemonmegas/common/CobblemonMegas.java:111]: You may want to add additional handling for this with your keystones, up to your discretion

File: common/src/main/java/com/selfdot/cobblemonmegas/common/command/MigrateMegaStone.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.selfdot.cobblemonmegas.common.item.MegaStoneHeldItemManager;
+import net.minecraft.item.ItemStack;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import net.minecraft.util.Hand;
+
+public class MigrateMegaStone implements Command<ServerCommandSource> {
```
Comment (landon-wills) [inline common/src/main/java/com/selfdot/cobblemonmegas/common/command/MigrateMegaStone.java:13]: Added this due to necessity on PokeClash to migrate player's megastones from previous mod, up to your discretion if you want to include this

File: common/src/main/java/com/selfdot/cobblemonmegas/common/command/CommandTree.java [hunk 3]
```diff
@@ -23,96 +23,100 @@
...
+                            return builder.buildFuture();
+                        })
+                        .then(RequiredArgumentBuilder.<ServerCommandSource, EntitySelector>
+                                        argument("players", EntityArgumentType.players())
+                                .executes(new GiveKeyStoneCommand())
+                        )
                 )
-            )
         );
+        dispatcher.register(LiteralArgumentBuilder.<ServerCommandSource>
```
Comment (landon-wills) [inline common/src/main/java/com/selfdot/cobblemonmegas/common/command/CommandTree.java:105]: Did not include permission node for this command as all users are expected to be able to use it, you may want to include one yourself

**Comment by davo899** [comment]
Thank you so much, really appreciate it.

File: common/src/main/java/com/selfdot/cobblemonmegas/common/command/MigrateMegaStone.java (hunk 2 above)
Comment (davo899) [reply common/src/main/java/com/selfdot/cobblemonmegas/common/command/MigrateMegaStone.java:13]: Absolutely, which mod does it migrate from?

File: common/src/main/java/com/selfdot/cobblemonmegas/common/command/CommandTree.java (hunk 3 above)
Comment (davo899) [reply common/src/main/java/com/selfdot/cobblemonmegas/common/command/CommandTree.java:105]: I think that's fine.

File: common/src/main/java/com/selfdot/cobblemonmegas/common/command/MigrateMegaStone.java (hunk 2 above)
Comment (davo899) [reply common/src/main/java/com/selfdot/cobblemonmegas/common/command/MigrateMegaStone.java:13]: Nevermind, just realised you said AaronMusk's.

File: common/src/main/java/com/selfdot/cobblemonmegas/common/CobblemonMegas.java (hunk 1 above)
Comment (davo899) [reply common/src/main/java/com/selfdot/cobblemonmegas/common/CobblemonMegas.java:111]: Do you know if there's an API for certain items being key items, or do I need to add and remove "cobblemon:key_stone" when the items are put in/taken from the player's inventory?

Comment (landon-wills) [reply common/src/main/java/com/selfdot/cobblemonmegas/common/CobblemonMegas.java:111]: I don't believe there is API for that yet. Looking at references to the keystones property shows only an extension function for retrieval.

If you want to add proper keystone logic, the route I would pursue first would be:

- Modify the `BattleStartedPreEvent` handler to add the identifier only if player has item in inventory
- Hook into the `BattleFledEvent` and `BattleVictoryEvent` and then remove identifiers

The one edge case I can think of here would be server crashing during battle and the player data becoming stale, so you would also want to remove the key in the `BattleStartedPreEvent` event if it's _not_ in the inventory but exists in the player data.

Comment (landon-wills) [reply common/src/main/java/com/selfdot/cobblemonmegas/common/CobblemonMegas.java:111]: Actually, I think either adding or removing (if it exists) on `BattleStartedPreEvent` depending on if it's in inventory should cover it? Shouldn't need to complicate it with the end-of-battle logic.

Comment (davo899) [reply common/src/main/java/com/selfdot/cobblemonmegas/common/CobblemonMegas.java:111]: Perfect, thanks.

**Pull request merged**